### PR TITLE
Mission file structure, refactoring, more dynamic settings.

### PR DESCRIPTION
**Pull request by theace0296**
I couldn't even describe every file that changed, but the basic idea was to refactor and reorganize the whole mission. Everything is a function of some sort now, with the exceptions of external libraries (AIS, COS, etc) and a few legecy files that redirect to their function counterparts (AddBadPoints, etc).

Feature changes/additions (That I can think of):
- Almost all globals are declared in fn_initGlobalVars and fn_initUnitVars, replacing the #includes in a lot of files.
- Added option for setting East, Ind, and West factions, available factions and their loadouts are automatically parsed from the ConfigFile.
- Added check before anything else gets done in fn_initGlobalVars for "laptop1", if it isn't found then the mission will run a series of functions to have the admin/sl pick a position for an HQ. Then the required objects are spawned in around that position. Not everything spawned has a 1 to 1 with the manually created missions, but the functionality is the same. Only required objects in mission file are Playable Blu Units, with the Civilian Playables and the HCs optional. 
- Added check in AIS for global variable "AIS_MOD_ENABLED", which allows anyone to quickly disable AIS. Specifically I usually run a mod version of AIS, so I don't want them to overwrite each other, but I still want TRGM to be able to call AIS functions.

I'm very aware that I didn't include every file from the master branch, and would like to include as much as I can now that the hard part is done. However, if you'd rather keep your current structure, then I'd be happy to just keep this refactored version as a fork for my own use.

### CODE REVIEW

**Comment by theace0296** [comment]
@treendy I left this as a draft so it could be tested more before a merge of any kind is done.

File: CustomMission/TRGMSetDefaultMissionSetupVars.sqf [hunk 1]
```diff
@@ -1,115 +0,0 @@
-
```
Comment (treendy) [inline CustomMission/TRGMSetDefaultMissionSetupVars.sqf:1]: This file is needed, see here: https://youtu.be/K6u34Ekxsmo?t=115
(have set the video to start at 1:55, as that is where i go into detail about the values)

File: CustomMission/TRGMSetEnemyFaction.sqf [hunk 2]
```diff
@@ -1,65 +0,0 @@
-
```
Comment (treendy) [inline CustomMission/TRGMSetEnemyFaction.sqf:1]: This file is needed, see here: https://youtu.be/K6u34Ekxsmo?t=115
(have set the video to start at 1:55, as that is where i go into detail about the values)

File: CustomMission/TRGMSetFriendlyLoadouts.sqf [hunk 3]
```diff
@@ -1,329 +0,0 @@
-
```
Comment (treendy) [inline CustomMission/TRGMSetFriendlyLoadouts.sqf:1]: This file is needed, see here: https://youtu.be/K6u34Ekxsmo?t=115
(have set the video to start at 1:55, as that is where i go into detail about the values)

**Review by treendy — COMMENTED** [review]
Hey, thanks for taking the time to go through all this :)
I will take a look and comment etc... (been meaning to move everything into functions)
Thank you :)

File: CustomMission/TRGMSetDefaultMissionSetupVars.sqf (hunk 1 above)
Comment (theace0296) [reply CustomMission/TRGMSetDefaultMissionSetupVars.sqf:1]: Slowly working it out, but see [here](https://github.com/treendy/TRGM2.Malden/pull/134/commits/d1084649ff41c0f8abaa7bcb34ad21ff150dc9d0?file-filters%5B%5D=.sqf#diff-88745d6c427a1bc1b88927173576e07aaaeafd47f84d50b2cb4d04e788663426).

**Comment by theace0296** [comment]
Since my version has evolved into something so different in function than the original TRGM2, it would be extremely hard to just "merge" the two together. For now, I'm breaking my fork into it's own repo, let me know if you would want to attempt to merge the two together at some point.

**Pull request closed without merging**